Plugins/CreateAPI/Plugin.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+import PackagePlugin
+
+@main
+struct Plugin: CommandPlugin {
+    func performCommand(context: PluginContext, arguments: [String]) async throws {
+        let createAPI = try context.tool(named: "create-api")
+        let currentDirectoryURL =  URL(fileURLWithPath: context.package.directory.string)
+            .appendingPathComponent("Sources")
+            .appendingPathComponent("OpenAPI")
+
+        let process = Process()
+        process.currentDirectoryURL = currentDirectoryURL
+        process.executableURL = URL(fileURLWithPath: createAPI.path.string)
+        process.arguments = [
+            "generate",
+            "app_store_connect_api_2.0_openapi.json",
+            "--module", "AppStoreConnect_Swift_SDK",
+            "--output", ".",
+            "--config", ".create-api.yml",
+            "--split"
+        ]
+
+        try process.run()
+        process.waitUntilExit()
+    }
```

Comment:
Just to touch back on https://github.com/CreateAPI/CreateAPI/issues/57#issuecomment-1200158360... In the future, you have access to a lot of info here that would allow you to do more advanced things such as iterate all of `context.package.target`'s and generate separate outputs for different targets.. Something like this:

```swift
@main
struct Plugin: CommandPlugin {
    func performCommand(context: PluginContext, arguments: [String]) async throws {
        for target in context.package.targets where target.directory.appending("Tags.txt").exists {
            let tags = target.directory.appending("Tags.txt").content.components(separatedBy: .newlines)

            // Call create-api with the given tags to generate...
        }
    }
}

extension Path {
    var exists: Bool {
        FileManager.default.fileExists(atPath: string)
    }

    var content: String {
        String(contentsOfFile: string, encoding: .utf8)
    }
}
```

Where the above assumes that you would list the array of tags that should be generated in a file called **Tags.txt**.

I'm sure there are a bunch of other approaches out there too, but just trying to give you an idea

Reply:
Thanks a lot, that definitely inspires!